### ISSUE TITLE
SDCard image does not boot on bpi-r4

### ISSUE THREAD

**Issue by BtbN** [frank-w/u-boot#16]
Both the image from the latest release here and one I built myself (for added EFI support, but I also tried building a vanilla one) just fail to boot with:

```
F0: 102B 0000
FA: 1042 0000
FA: 1042 0000 [0200]
F9: 1041 0000
F3: 1001 0000 [0200]
F3: 1001 0000
F6: 380E 5800
F5: 4801 0000
00: 1005 0000
FA: 1042 0000
FA: 1042 0000 [0200]
F9: 1041 0000
F3: 1001 0000 [0200]
F3: 1001 0000
F6: 380E 5800
01: 102A 0001
02: 1005 0000
BP: 0200 00C0 [0001]
EC: 0000 0000 [3000]
MK: 0000 0000 [0000]
T0: 0000 0112 [0101]
System halt!
```

If I flash an sdcard image from the openwrt snapshots (https://firmware-selector.openwrt.org/?version=SNAPSHOT&target=mediatek%2Ffilogic&id=bananapi_bpi-r4), the system boots just fine. So I'm pretty confident the SD card, the hardware and my method of writing the image is okay.

**Comment by frank-w** [comment]
Are you sure you used the right file (which one exactly)? Have not changed atf/image part for long time and not even bl2 can be loaded

**Comment by BtbN** [comment]
Found the issue.
I ended up with two solid files, /dev/loop0p1 and /dev/loop0p4 in place of the correct dev nodes.
So those two files never made it to the SD card, since losetup did not replace them.
After deleting them and re-running createimg, it works fine.

I'm not sure why the downloaded image didn't work fine. Maybe it failed to flash correctly, due to the substantially larger size (I shrunk mine down to 250MB)
